### DISCUSSION TITLE
Midi CC control of top level drum FX / song-level FX

### DISCUSSION THREAD

**Discussion by mattyharm** [SynthstromAudible/DelugeFirmware#2016] [Ideas]
Apologies if someone's already got this as an open request, but it'd be great to get to a point where midi control of the "effect all" layer of effects (cutoff, etc) on kits and the "effect all" layer of effects on the whole song were possible.  I know it's probably not the easiest thing to add since those layers don't have a way to access them on the OLED screen, and that's the usual way to learn external CC controls, but I just wanted to put it out there.

**Comment by sapphire-arches** [comment]
Both of these work today, at least in 1.1. To learn kit affect-all params, make sure affect-all is enabled while in a kit clip, press `SELECT`, navigate to the menu for the parameter you want to learn, then hold `LEARN/INPUT`. The same process works for songs, just starting from a song view.

**Comment by mattyharm** [comment]
Welp I feel dumb!  Thank you, I had no idea!

**Reply by sapphire-arches** [reply]
no worries, it's not easy to discover 😅
